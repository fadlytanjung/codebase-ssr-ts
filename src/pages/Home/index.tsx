import React, { useState } from 'react';
import Button from '@/components/elements/Button';
import ButtonReact from '@/components/elements/ButtonEmotion';
import ButtonStyled from '@/components/elements/ButtonTailwind';
import styles from './styles.module.css';

function HomePage(props: { title: string }) {
  const [count, setCount] = useState(0);

  return(
    <div className={styles.root}>
      <Button onClick={()=> setCount(count+1)} variant="primary">Add + </Button>
      <ButtonReact className={styles.overrideBtn}>Button @emotion/react</ButtonReact>
      <ButtonStyled className={styles.overrideBtn}>Button @emotion/styled</ButtonStyled>
      <h1>Total count: {count}</h1>
      <p>Welcome, {props.title}</p>
    </div>
  );
}

export default HomePage;

