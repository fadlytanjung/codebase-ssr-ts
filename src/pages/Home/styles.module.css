.overrideBtn{
  min-width: 12rem !important;
}

.root > button {
  margin-bottom: 1rem;
}