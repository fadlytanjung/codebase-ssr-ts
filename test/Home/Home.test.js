import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Home from '@/pages/Home';

describe('Test Home Page', () => {
  test('It should render with snapshot', async () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should find text', async () => {

    const { findByText, container } = render(<Home title="Fadly" />);
    const addBtn = await findByText('Add +');
    const h1Text = container.querySelector('h1');

    fireEvent.click(addBtn);
    expect(h1Text).toHaveTextContent('Total count: 1');
  });
});
