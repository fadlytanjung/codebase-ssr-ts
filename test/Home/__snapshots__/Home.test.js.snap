// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page It should render with snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Add + 
    </button>
    <h1>
      Total count: 0
    </h1>
    <p>
      Welcome, 
    </p>
  </div>
</DocumentFragment>
`;
